lib/debug/repl.cpp

```diff
@@ -700,6 +700,46 @@ void Repl::addTemperatureCommands(const std::uint8_t bus, const std::uint8_t dev
   addCommand(temperature_read_command);
 }
 
+void Repl::addImdCommands(const std::string &bus)
+{
+  const auto optional_can = getCan(bus);
+  if (!optional_can) {
+    logger_.log(core::LogLevel::kFatal, "Failed to create CAN instance on bus %s", bus.c_str());
+    return;
+  }
+  const auto can          = std::move(*optional_can);
+  const auto optional_imd = sensors::Imd::create(logger_, can);
+  if (!optional_imd) {
+    logger_.log(core::LogLevel::kFatal, "Failed to create IMD instance");
+    return;
+  }
+  auto imd = std::move(*optional_imd);
+  Command imd_request_values;
+  imd_request_values.name = "imd request values";
+  imd_request_values.description
+    = "Request values from the IMD for positive and negative resistances, and isolation state";
+  imd_request_values.handler = [this, imd, bus, can]() {
+    core::Result result = imd->updateValues();
+    if (result == core::Result::kFailure) {
+      logger_.log(core::LogLevel::kFatal, "Failed to run IMD::updateValues()");
+      return;
+    }
+    sleep(1);
```

Comment:
Was advised by @TomLonergan03 so that enough time is given to allow CAN device to receive message and respond with its own. If there's a better way of doing this though, let me know

Reply:
yeah for the repl this is the best way I've worked out to trigger `can_.receive()` and guarantee that the message has been returned. tbh you could probably immediately run `can_.receive()` but I'm not 100% sure.
in the actual code `can_.receive()` will be called in one of the main loops so will be dealt with asynchronously (relative to message send at least) hence why we have to sleep